api/lib/opentelemetry.rb

```diff
@@ -22,13 +22,27 @@
 module OpenTelemetry
   extend self
 
-  attr_writer :tracer_provider, :meter_provider, :baggage, :logger
+  attr_writer :tracer_provider, :meter_provider, :baggage, :logger, :error_handler
 
   # @return [Object, Logger] configured Logger or a default STDOUT Logger.
   def logger
     @logger ||= Logger.new(STDOUT, level: ENV['OTEL_LOG_LEVEL'] || Logger::INFO)
   end
 
+  # @return [Callable] configured error handler or a default that logs the
+  #   exception and message at ERROR level.
+  def error_handler
+    @error_handler ||= ->(exception: nil, message: nil) { logger.error("OpenTelemetry error: #{[message, exception&.message].compact.join(' - ')}") }
+  end
+
+  # Handles an error by calling the configured error_handler.
+  #
+  # @param [optional Exception] exception The exception to be handled
+  # @param [optional String] message An error message.
+  def handle_error(exception: nil, message: nil)
+    error_handler.call(exception: exception, message: message)
```

Comment:
I like this, it provides a simple flexible interface to let users to configure how they want to handle exceptions, or non-exceptional errors (like invalid attributes)